helper/locking/lock.go

```diff
@@ -1,19 +1,46 @@
-//go:build !deadlock
-
 package locking
 
 import (
 	"sync"
+
+	"github.com/sasha-s/go-deadlock"
 )
 
-// DeadlockMutex is just a sync.Mutex when the build tag `deadlock` is absent.
-// See its other definition in the corresponding deadlock-build-tag-constrained
-// file for more details.
+// Common mutex interface to allow either built-in or imported deadlock use
+type Mutex interface {
```

Comment:
There's nothing wrong with this interface based approach.  I will note however that there now appears to be a way to disable deadlock detection via https://github.com/sasha-s/go-deadlock (see the end of the README), so that's another, possibly simpler, approach we could consider.

Reply:
spoke with @ncabatoff about this, and we decided that the current implementation is best because opts are global in `go-deadlock` https://github.com/sasha-s/go-deadlock/blob/master/deadlock.go#L17. We want to be able to turn deadlock detection on for somethings, but not everything.